doc/api_reference.md

```diff
@@ -210,6 +212,36 @@ If any of the input senders complete with done or error then it will request
 any senders that have not yet completed to stop and the operation as a whole
 will complete with done or error.
 
+### `materialize(Sender sender) -> Sender`
+
+Materializes the completion signal of `sender` into the value-channel by
+invoking prepending the completion arguments with the corresponding
+`set_value`, `set_error` or `set_done` CPO as an additional argument.
+
+ie. Transforms the following LHS completion signals to the RHS completion signals
+* `set_value(r, values...)` -> `set_value(r, set_value, values...)`
+* `set_error(r, e)` -> `set_value(r, set_error, e)`
+* `set_done(r)` -> `set_value(r, set_done)`
+
+This allows you to treat any result as a success and process the result as
+a value.
```

Comment:
This could be used in conjunction with something like `when_all()` to allow you to inspect each of the results without having the failure/cancellation of one operation resulting in a failure of the entire operation.